IPIP/0000-delegated-routing-http-api.md

```diff
@@ -0,0 +1,88 @@
+The cost of a proper redesign, implementation, and maintenance of Reframe and its implementation is too high relative to the urgency of having a delegated routing HTTP API.
+
+Note that this does not supplant nor deprecate Reframe. Ideally in the future, Reframe and its implementation would receive the resources needed to map the IDL to idiomatic HTTP,
+and this spec could then be rewritten in the IDL, maintaining backwards compatibility.
+
+## Detailed design
+
+See the [API design](../routing/DELEGATED_ROUTING_HTTP.md) included with this IPIP.
+
+## Design rationale
+To understand the design rationale, it is important to consider the concrete Reframe limitations that we know about:
+
+- Reframe methods are encoded inside messages
+    - This prevents URL-based pattern matching on methods
+        - Configuring different caching strategies for different methods
+        - Configuring reverse proxies on a per-method basis
+            - Routing methods to specific backends
+            - Method-specific reverse proxy config such as timeouts
+    - Developer UX is poor as a result, e.g. for CDN caching you must encode the entire request message and pass it as a query parameter
+        - This was initially done by URL-escaping the raw bytes
+          - Not possible to consume correctly using standard JavaScript
+          - Shipped in Kubo 0.16
+        - Packing a CID into a struct, encoding it with DAG-CBOR, multibase-encoding that, percent-encoding that, and then passing it in a URL, rather than merely passing the CID in the URL, is needlessly complex from a user's perspective
+        - Added complexity of "Cacheable" methods supporting both POSTs and GETs
+- The required streaming support and message groups add a lot of implementation complexity but isn’t very useful
+    - Ex for FindProviders, the response is buffered anyway for ETag calculation
+    - There are no limits on response sizes nor ways to impose limits and paginate
+    - This is useful for routers that have highly variable resolution time, to send results as soon as possible, but this is not a use case we are focusing on right now and we can add it later
+- The Identify method is not implemented because it is not currently useful
+- Client and server implementations are difficult to write correctly, because of the non-standard wire formats and conventions
+- The Go implementation is [complex](https://github.com/ipfs/go-delegated-routing/blob/main/gen/proto/proto_edelweiss.go) and [brittle](https://github.com/ipfs/go-delegated-routing/blame/main/client/provide.go#L51), and is currently maintained by IPFS Stewards who are already over-committed with other priorities
+- Only the HTTP transport has been designed and implemented, so it's unclear if the existing design will work for other transports, and what their use cases and requirements are
+
+So this API proposal makes the following changes:
```

Comment:
I would add a specific URL prefix like `/delegated/v1/....` as mandatory to avoid DoS attacks.

Reply:
because if we do not set a specific one, depending on the use case (see https://github.com/ipfs/specs/pull/322) this URL can be provided as a hint. If the content is very popular, we are going to hit that endpoint from thousands of clients. If we limit delegated routing endpoints to have a prefix, then we can avoid that behavior by not reaching that endpoint if it does not start with `/drouting` prefix or similar.